// graphql-cli/dist/cli.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const yargs_1 = tslib_1.__importDefault(require("yargs"));
const globby_1 = tslib_1.__importDefault(require("globby"));
const path_1 = require("path");
const common_1 = require("@test-graphql-cli/common");
async function cli() {
    const program = yargs_1.default
        .scriptName('graphql')
        .detectLocale(false)
        .epilog('Visit https://github.com/Urigo/graphql-cli for more information')
        .version();
    const commandPackageNames = await discoverCommands();
    const commandFactories = await Promise.all(commandPackageNames.map(loadCommand));
    commandFactories.forEach((cmd) => {
        program.command(cmd({
            useConfig: common_1.useConfig,
        }));
    });
    program.demandCommand().recommendCommands().help().showHelpOnFail(false).argv;
}
exports.cli = cli;
async function discoverCommands() {
    const discoveredCommands = await globby_1.default('*', {
        cwd: path_1.join(process.cwd(), 'node_modules/@test-graphql-cli'),
        onlyDirectories: true,
        deep: 1,
        ignore: ['common'],
    });
    return discoveredCommands;
}
function loadCommand(name) {
    const mod = require(`@test-graphql-cli/${name}`);
    return mod.default || mod;
}
